pkg/objstore/swift/swift.go

```diff
@@ -82,215 +190,130 @@ func (c *Container) Name() string {
 
 // Iter calls f for each entry in the given directory. The argument to f is the full
 // object name including the prefix of the inspected directory.
-func (c *Container) Iter(ctx context.Context, dir string, f func(string) error) error {
-	// Ensure the object name actually ends with a dir suffix. Otherwise we'll just iterate the
-	// object itself as one prefix item.
+func (c *Container) Iter(_ context.Context, dir string, f func(string) error) error {
 	if dir != "" {
-		dir = strings.TrimSuffix(dir, DirDelim) + DirDelim
+		dir = strings.TrimSuffix(dir, string(DirDelim)) + string(DirDelim)
 	}
-
-	options := &objects.ListOpts{Full: true, Prefix: dir, Delimiter: DirDelim}
-	return objects.List(c.client, c.name, options).EachPage(func(page pagination.Page) (bool, error) {
-		objectNames, err := objects.ExtractNames(page)
+	return c.connection.ObjectsWalk(c.name, &swift.ObjectsOpts{Prefix: dir, Delimiter: DirDelim}, func(opts *swift.ObjectsOpts) (interface{}, error) {
+		objects, err := c.connection.ObjectNames(c.name, opts)
 		if err != nil {
-			return false, err
+			return objects, errors.Wrap(err, "swift list object names")
 		}
-		for _, objectName := range objectNames {
-			if err := f(objectName); err != nil {
-				return false, err
+		for _, object := range objects {
+			if object == SegmentsDir {
+				continue
+			}
+			if err := f(object); err != nil {
+				return objects, errors.Wrap(err, "swift iteration over objects")
 			}
 		}
-
-		return true, nil
+		return objects, nil
 	})
 }
 
-// Get returns a reader for the given object name.
-func (c *Container) Get(ctx context.Context, name string) (io.ReadCloser, error) {
+func (c *Container) get(name string, headers swift.Headers, checkHash bool) (io.ReadCloser, error) {
 	if name == "" {
-		return nil, errors.New("error, empty container name passed")
+		return nil, errors.New("Object name cannot be empty")
```

Comment:
```suggestion
		return nil, errors.New("object name cannot be empty")
```

Reply:
resolved in https://github.com/thanos-io/thanos/pull/2732/commits/7a51f585a7184cc9a0c5a73ffcd77b23896da026